### ISSUE TITLE
Unexpected Component Lifecycle on LWC OSS in native Shadow DOM mode

### ISSUE THREAD

**Issue by jmpisson** [salesforce/lwc#1452]
### Description
In native shadow DOM mode, the component lifecycle of slot content and the slot receiver is not running in the sequence expected.

#### Steps to Reproduce
Considering my-app as the Root Component inside LWC project, we have two components, Parent and Child, in the app markup:

my/app/app.html
````html
<template>
    <div>
        <my-parent>
                <my-child></my-child>
        </my-parent>
    </div>
</template>
````

my/parent/parent.html
````html
<template>
    <div>
        <h2>Parent</h2>
        <slot></slot>
    </div>
</template>
````

my/parent/parent.js
````js
import { LightningElement } from "lwc";

export default class Parent extends LightningElement {
    connectedCallback() {
        console.log('Connected Parent');
    }

    renderedCallback() {
        console.log('Rendered Parent');
    }
}
````

my/child/child.html
````html
<template>
    <div>
        <h2>Child</h2>
    </div>
</template>
````

my/child/child.js
````js
import { LightningElement } from "lwc";

export default class Child extends LightningElement {
    connectedCallback() {
        console.log('Connected Child');
    }

    renderedCallback() {
        console.log('Rendered Child');
    }
}
````
Load the page and see the JavaScript Console Output

### Playground link
Run with Native Shadow DOM option selected
https://playground.lwcjs.org/projects/3-XFTm3xZ/2/edit

#### Expected Results
Console Output:
````
Connected Parent
Connected Child
Rendered Child
Rendered Parent
````

Expected behavior in native web components: https://jsbin.com/vavenef/edit?html,js,console
```
Connected Parent
Connected Child
```
#### Actual Results
Console Output:
````
Connected Child
Rendered Child
Connected Parent
Rendered Parent
````

### Browsers Affected
Firefox 68.173.0.3683.75 built on Debian buster/sid, running on Debian 10.0 (64-bit)
Chromium 73.0.3683.75 (Developer Build) built on Debian buster/sid, running on Debian 10.0 (64-bit)
Gnome Web 3.32 built on Debian buster/sid, running on Debian 10.0 (64-bit)

### Version
Node project
- package.json
  "lwc-services": "^1.2.2"
- LWC: 1.0.0

**Comment by caridy** [comment]
cc @jodarove can you take a look at this?

**Comment by caridy** [comment]
@jmpisson there might be something going on here... let me loop in @ravijayaramappa who is working on this area at the moment. There are a couple of things:

* the virtual dom library does some optimizations when appending elements, usually works from bottom to top.
* the rendered phase is something that we, the LWC engine, will control, we can decide to do render on connect, or render on the microtask, so, in that case, it is not deterministic.

**Comment by nolanlawson** [comment]
AIUI this issue is fixed, and may be further fixed by #3198. Closing for now, please reopen as necessary.
